sysconfig/cloudinit.go

```diff
@@ -743,6 +744,14 @@ func CloudInitStatus() (CloudInitState, error) {
 		return CloudInitDisabledPermanently, nil
 	}
 
+	// if it was explicitly disabled via the kernel commandline, then
+	// return special status for that, if commandline is unavailable,
+	// assume not disabled
+	cmdline, err := kcmdline.KeyValues("cloud-init")
+	if err == nil && cmdline["cloud-init"] == "disabled" {
+		return CloudInitDisabledPermanently, nil
+	}
```

Comment:
I think I'd log something if kernel command line cannot be parsed (it's unlikely we'd even reach this far in such case, but who knows):
```suggestion
	cmdline, err := kcmdline.KeyValues("cloud-init")
	if err != nil {
		logger.Noticef("cannot obtain cloud-init from kernel command line: %v", err)
	} else if err == nil && cmdline["cloud-init"] == "disabled" {
		return CloudInitDisabledPermanently, nil
	}
```

Reply:
Sure, seems reasonable to me.